### PR TITLE
Documentation and Examples

### PR DESCRIPTION
**Pull request by jacobo**
Redis optional cookbook with example

Hello World custom cookbook example

Documentation on custom cookbooks

### CODE REVIEW

File: cookbooks/redis/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Overview
+
+This cookbook provides a method to host a dedicated redis instance.  This recipe should *NOT* be used on your Database instance as it is not designed for that instance.  Additionally it will not change nor modify your database instance in anyway what so ever.
+
+## Installation
+
+Since this is an optional recipe, it should be installed by simply including it via a `depends` in your `ey-custom/metadata.rb` file and an `include_recipe` in the appropriate hook file. For more details on optional recipes see the [redis example]. (Yes, how convenient that the example is for the exact recipe you wanted)
+
+Additionally, this recipe will only activate on insta
```
Comment (crigor) [inline cookbooks/redis/README.md]: Typo on insta

File: cookbooks/redis/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+```ruby
+default['redis'] = {
+  'utility_name' => "my_custom_name", # default: redis
+  # ...
+}
+```
+
+Ensure you have the Dependencies installed in your local cookbooks repository ...
+Add the following to your main/recipes/default.rb
```
Comment (radamanthus) [inline cookbooks/redis/README.md]: The reference to `main/recipes/default.rb` is my fault, carried over from https://github.com/engineyard/ey-cookbooks-stable-v5-DELETEME/blob/master/cookbooks/redis/README.md. 

I'll put in a PR to fix this as soon as this PR is merged.

**Comment by jacobo** [comment]
@radamanthus I've updated the `README`, does that look like appropriate instructions now?

**Comment by radamanthus** [comment]
Looks good!

**Comment by jacobo** [comment]
ready to merge

File: cookbooks/ey-init/recipes/main.rb [hunk 3]
```diff
@@ -15,11 +15,15 @@
...
+Chef::Log.info("Calling custom recipes hook: before-main")
 include_recipe 'ey-custom::before-main'
+Chef::Log.info("Finished custom recipes hook: before-main")
 
 include_recipe 'ey-base::bootstrap' # common things that are installed in all the instances
 node.engineyard.instance.roles.each { |role| include_recipe "#{role}::prep" }
 node.engineyard.instance.roles.each { |role| include_recipe "#{role}::build" }
 include_recipe 'ey-base::post_bootstrap' # common things that we want to install setting up the instance
 
+Chef::Log.info("Calling custom recipes hook: after-main")
```
Comment (aespinosa) [inline cookbooks/ey-init/recipes/main.rb]: I think this gets logged at compile time.  Use https://docs.chef.io/resource_log.html instead to log things properly on the converge.

Comment (jacobo) [inline cookbooks/ey-init/recipes/main.rb]: let's not log for now then and just get this merged...

**Comment by tpoland** [comment]
LGTM

**Pull request merged**